smartsim/_core/entrypoints/colocated.py

```diff
@@ -203,37 +207,51 @@ def main(
 
     try:
         logger.debug(
-            "\n\nColocated database information\n"
-            + "\n".join(
+            ("\n\nColocated database information\n"
+            "\n".join(
                 (
-                    f"\tIP Address(es): {' '.join(ip_addresses + [lo_address])}",
-                    f"\tCommand: {' '.join(cmd)}\n\n",
-                    f"\t# of Database CPUs: {db_cpus}",
+                    "\tIP Address(es): {}"
+                    "\tCommand: {}\n\n",
+                    "\t# of Database CPUs: {}",
                 )
             )
+            ),
+            ' '.join(ip_addresses + [lo_address]),
+            ' '.join(cmd),
+            db_cpus
```

Comment:
I could be wrong, but I don't think the new code behaves the same as the old

Reply:
I see a missing comma on line 213. Fantastic catch. I hate this construction with the joins, altogether. Replacing with simpler version that doesn't unnecessarily use the join:

```
logger.debug(
    (
        "\n\nColocated database information\n"
        "\n\tIP Address(es): {}"
        "\n\tCommand: {}\n\n"
        "\n\t# of Database CPUs: {}"
    ),
    ' '.join(ip_addresses + [lo_address]),
    ' '.join(cmd),
    db_cpus
)
```